filelog_unix.go

```diff
@@ -2,4 +2,69 @@
 
 package log4go
 
+import (
+	"compress/gzip"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+)
+
 const FILELOG_ARCHIVE_REGEX = `\.[0-9]{4}-[0-9]{2}-[0-9]{2}(\.[0-9]{4})?(\.gz)?$`
+
+func (w *FileLogWriter) compressFile(filename string) {
+	basename := filepath.Base(filename)
+	dirname := filepath.Dir(filename)
+
+	plainFile, err := os.Open(filename)
+	if err != nil {
+		fmt.Fprintf(w.errorWriter, "FileLogWriter(%q): Couldn't open logfile to begin compression: %s\n", w.filename, err)
+		return
+	}
+
+	compressedFilename := filepath.Join(dirname, basename+".gz")
```

Comment:
Its sorta wild, but that's how spaces in `gofmt` work: https://github.com/golang/go/issues/17196. The second argument is treated differently than the first.